src/posix/tty.rs

```diff
@@ -105,6 +105,11 @@ impl TTYPort {
             nix::sys::stat::Mode::empty(),
         )?);
 
+        // Try to claim exclusive access to the port. This is performed even
+        // if the port will later be set as non-exclusive, in order to respect
+        // other applications that may have an exclusive port lock.
+        ioctl::tiocexcl(fd.0)?;
+
```

Comment:
This is a problem on custom TTY device implementations that do not support this operation, which makes using this library without a patch on those TTYs impossible.